src/util/strencodings.h

```diff
@@ -305,4 +305,17 @@ std::string ToUpper(const std::string& str);
  */
 std::string Capitalize(std::string str);
 
+/**
+ * Parse a string with suffix unit [k|K] [m|M] [g|G] [t|T] .
+ * Must be a whole integer, fractions not allowed (0.5t).
+ * Lowercase units are 1000 base. Uppercase units are 1024 base.
+ * Examples: 2m,27M,19g,41T
+ *
+ * @param[in] str                  the string to convert into bytes
+ * @param[in] default_multiplier   if no unit is found in str, use this multiplier. default is MiB
+ * @returns                        optional uint64_t bytes from str or nullopt
+ *                                 if ParseUInt64 is false, str is empty, trailing whitespace or overflow
+ */
+std::optional<uint64_t> ParseByteUnits(const std::string& str, uint64_t default_multiplier = 1024UL * 1024UL);
```

Comment:
Should the default multiplier have a default? It seems like the default is something that is per-arg and not project-wide.

Reply:
Agreed.  I remove the default.

Reply:
How about `1` as a default multiplier ?

Reply:
How about removing the argument? It's not needed. Either way please update doc comment above.

Reply:
default of `1` seems natural, so my latest change defaults to `1`

Reply:
> How about removing the argument? It's not needed.

Indeed! I would remove that unused code. No parameter, no problem :) (what should be its default value? should it have default value? how to test `0` if we have assert()?)

Reply:
i do need this multiplier argument. the intent of this function is to be used with maxupload, dbcache...etc. they inheritly have a default multiplier (mostly MiB).  the main purpose is maintain backwards capability, e. g. `maxupload=300`  becomes `300m`.
As I write this, i think I will change the argument to a an enum with the valid units [k|K|m|M...]
with no default value. This makes it clear and easy to verify/test.

Reply:
@vasild @MarcoFalke @promag 
I added an enum `ByteUnitMultiplier` to [src/util/strencodings.src/util/strencodings.h](https://github.com/bitcoin/bitcoin/pull/23249/files#r748048958) 
Let me know what you think.